src/canopy/llm/models.py

```diff
@@ -32,11 +32,12 @@ class FunctionParameters(BaseModel):
     required_properties: List[FunctionProperty]
     optional_properties: List[FunctionProperty] = []
 
-    def dict(self, *args, **kwargs):
+    @model_serializer()
```

Comment:
Why did you replace `def dict()` sometimes with `def model_dump` and sometimes with `@model_serializer`?  
Is there a difference between the two? If not, it's probably a good idea to maintain a single code style for all models

Reply:
The only difference is that you cannot pass args and kwargs to model_serializer. In any case I changed to model serializers since we don't use these args for now.

Reply:
Actually it failed the tests rolling back I get:

pydantic_core._pydantic_core.PydanticSerializationError: Error calling function `serialize_model`: RecursionError which kinda makes sense